### PR TITLE
Add front-end section

### LINKED ISSUE #168: Front-end figures and text
I'll put some figures and write-ups on my involvement in the Manubot project in this issue, so people can review theme for inclusion in the Meta Review.

Here's the first one, overviewing the plugins:

plugins

I'll probably do another figure of just a general screenshot of the theme and plugins together (more of an overview screenshot).

### PR DESCRIPTION
**Pull request by vincerubinetti**
Closes #168

### CODE REVIEW

File: content/02.main-text.md
Comment (dhimmel) [inline content/02.main-text.md]: `(Figure @fig:plugins)`

File: content/02.main-text.md
Comment (dhimmel) [inline content/02.main-text.md]: Perhaps there is some way we can note that these plugins/themes assumes pandoc-generated HTML and thus can be used by the large body of projects that use pandoc.

File: content/02.main-text.md
Comment (dhimmel) [inline content/02.main-text.md]: This is a bad sentence, but do we want to add something like:

> For example, citation tooltips provide users with access upon hovering to the reference and enable navigated to all other instances of that a given citation.

I think that commenting on at least one of the panels could help users approach this figure.

**Review by dhimmel — COMMENTED** [review]
Should we mention that the plugins are somewhat configurable?

File: content/02.main-text.md
Comment (vincerubinetti) [reply content/02.main-text.md]: We could say that. I'd want to test it on some other just-pandoc manuscripts before claiming that though. There are certain assumptions it makes, like the fignos filters, and the --section parameter, and things like that.

File: content/02.main-text.md
Comment (vincerubinetti) [reply content/02.main-text.md]: How about this:

> For example, with the "tooltips" plugin enabled, when the user hovers over a link to a reference or figure, a preview of that item pops up over the link, along with controls to navigate between other mentions of that item elsewhere in the document.

I'd put it right after the first sentence.

File: content/02.main-text.md
Comment (dhimmel) [reply content/02.main-text.md]: Makes sense. So let's hold off on this assertion until we have time to evaluate it. (which may not be until after submitting the meta review).

**Comment by vincerubinetti** [comment]
Added a sentence and tweaked some wording.

File: content/02.main-text.md
Comment (agitter) [inline content/02.main-text.md]: "authors" sounds slightly better to me than "the author"

File: content/02.main-text.md
Comment (agitter) [inline content/02.main-text.md]: Should we way "most major web browsers"?

**Review by agitter — CHANGES_REQUESTED** [review]
Great section.

In addition to the phrasing comments, should we add the source file for `images/plugins.png` to the repository?  We have tried to commit or link to the original file so that any author can edit images later if needed.

We will need to update the response to reviewers as well, but could handle that in a separate pull request.

File: content/02.main-text.md
Comment (dhimmel) [reply content/02.main-text.md]: > For example, with the "tooltips" plugin enabled, when the user hovers over a link to a reference or figure, a preview of that item pops up over the link, along with controls to navigate between other mentions of that item elsewhere in the document.

This still needs to be added.

Also we should probably attribute the source of the screenshots:

```markdown
All screenshots show snippets from the [Sci-Hub Coverage Study](https://greenelab.github.io/scihub-manuscript/v/fd7acb7ed0108c920da56f84819ce13f02f68aa8/) besides for the Hypothesis screenshot, which was taken from the [TPOT-DS manuscript](https://trang1618.github.io/tpot-ds-ms/).
Both manuscripts are available under a [CC BY 4.0 License](https://creativecommons.org/licenses/by/4.0/ "Creative Commons Attribution 4.0 International License").
```

Comment (vincerubinetti) [reply content/02.main-text.md]: I did add the example tooltip sentence. It's showing up for me in the diff.

I can put the screenshot attribution in the figure caption? Is it necessary to state the license in this manuscript, when they can follow the links to the other manuscripts if they need that info? Idk if it's important enough to list here?

Comment (dhimmel) [reply content/02.main-text.md]: > I did add the example tooltip sentence. It's showing up for me in the diff.

Ah I see. I was envisioning the sentence would go in the caption, since I think it's most useful for users when they are directly viewing the figure. Not a strong preference however.

> Is it necessary to state the license in this manuscript

Yes, it is best practice. As per the non-legal (more readable) CC BY license terms:

> Attribution — You must give appropriate credit, **provide a link to the license**, and indicate if changes were made. You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.

Bolding is mine. The actual license text states:

> indicate the Licensed Material is licensed under this Public License, and include the text of, or the URI or hyperlink to, this Public License.

We might also want to state in the caption that our markup is in purple, not just for the CC BY modifications clause compliance, but also so there is no uncertainty.

Comment (vincerubinetti) [reply content/02.main-text.md]: I've added it.

Comment (dhimmel) [reply content/02.main-text.md]: Great! Can you bold the first sentence of the caption? In other contexts, this will become the figure caption title.

**Review by dhimmel — APPROVED** [review]
Looks good to me.

Guessing we'll respond to reviewer comment in a subsequent PR.

@agitter ready for your re-review.

**Review by agitter — COMMENTED** [review]
The text looks good to me.  I have two more questions about the figure:
- Should we use a gh-pages hosted manuscript for the anchor example URL, or is this not important?
- Because most of the examples come from the Sci-Hub manuscript, a careless reader may get the impression that Manubot or this paper are somehow about Sci-Hub.  The citation tooltips panel is especially relevant here because the **pirated papers** title is one of the focal points of the entire figure when I glance at it.  Do we want to draw some examples from deep review, which is one of the subjects of this paper?

**Comment by dhimmel** [comment]
> Should we use a gh-pages hosted manuscript for the anchor example URL, or is this not important?

Good catch. I think it would be preferable to use a public manuscript URL.

> Because most of the examples come from the Sci-Hub manuscript, a careless reader may get the impression that Manubot or this paper are somehow about Sci-Hub

I think that most panel text is rather small and inconsequential. However, the bibliographic references do stick out to me. We could pick different references from the Sci-Hub Coverage Study. However, it's also perhaps a good idea to add 1 more manuscript to the mix.

**Comment by agitter** [comment]
How about `[150]` from this [deep review section](https://greenelab.github.io/deep-review/#continuous-collaborative-manuscript-drafting) as a new example citation?  It's on topic.

image

**Comment by vincerubinetti** [comment]
> > Should we use a gh-pages hosted manuscript for the anchor example URL, or is this not important?
> 
> Good catch. I think it would be preferable to use a public manuscript URL.

I disagree. The point of the screenshots is to draw attention to the particular feature of the plugin, and nothing else (or as little else as possible). I purposely tried to think of the most generic url possible. We shouldn't distract from the main point of the screenshot. I'd even consider just making it `manuscript.html#methods` as to be the least confusing.

___

Regarding the other things: I don't think anyone will look closely enough to realize they're all from the same paper. They'll just see it as example text. However, I could certainly choose a different tooltip other than the "pirated papers" one. Perhaps people don't have a positive association with that word.

We could add the word "example" to the figure caption text to attempt to make it clear that Manubot isn't inherently tied to the SciHub paper.

> However, it's also perhaps a good idea to add 1 more manuscript to the mix.

Another manuscript adds another bulky phrase and link we have to add regarding its license. The less extraneous info to distract the reader, the better imo. The point of the figure is to show the functionality of the plugins, not to show the diversity of manuscripts that have already been written using Manubot (if you want to show of that, it should go elsewhere).

**Comment by agitter** [comment]
>  However, I could certainly choose a different tooltip other than the "pirated papers" one. Perhaps people don't have a positive association with that word.

That's part of what I had in mind.  I know the Sci-Hub study and its value, but others at a glance will not know the context and may perceive a negative connotation with "pirated".

> Another manuscript adds another bulky sentence we have to add regarding its license.

I agree we do not want to distract readers from the point of the figure.  Could we shorten the caption text:

> All screenshots were taken manuscripts [X, Y, Z] available under a [CC BY 4.0 License](https://creativecommons.org/licenses/by/4.0/ "Creative Commons Attribution 4.0 International License").

where `[X, Y, Z]` are standard citations to these three manuscripts (SciHub, Deep Review, TPOT-DS) instead of their names?  If we want to track the specific permalink, that could be included as a comment.

**Comment by vincerubinetti** [comment]
I found this photogenic tooltip that also happens to have all the citations in the same paragraph. That way if someone does happen to look closely, they'll see "1 of 3" and be able to see all 3 citations in the screenshot.
tooltips

**Comment by agitter** [comment]
@vincerubinetti that new citation screenshot looks nice to me.  If you don't want to add deep review to our source manuscripts, I'm okay proceeding with this one.  I still recommend shorting the caption text to something like

> All screenshots were taken manuscripts [X, Y] available under a CC BY 4.0 License.

even if we only have two source manuscripts.

I don't have a strong opinion about the anchor example URL.  Let's see what @dhimmel thinks.

**Comment by vincerubinetti** [comment]
@agitter Spoke with @dhimmel. He wanted to link to the Manubot versions of those manuscripts instead of the DOI/elife versions that are actually in the references. Rather than adding a duplicate entry for the Manubot versions in the references, he's opting to just link to them directly (without "citing" them, I guess you could say).

**Comment by dhimmel** [comment]
In short, I though it was important for us to direct users to a webpage where they can actually see the frontend depicted in the screenshots. Therefore, citing the journal article is not a sufficient solution, although we could do that in addition.

File: content/02.main-text.md
Comment (dhimmel) [inline content/02.main-text.md]: I would add `are` to make this a sentence. Same as above, I would add `were` before taken. I think the convention is that the caption title doesn't need to be a sentence, but the remainder should be sentence based. Not a huge deal probably.

**Comment by dhimmel** [comment]
> I disagree. The point of the screenshots is to draw attention to the particular feature of the plugin, and nothing else (or as little else as possible). I purposely tried to think of the most generic url possible. We shouldn't distract from the main point of the screenshot. I'd even consider just making it manuscript.html#methods as to be the least confusing.

I like `manuscript.html#methods` as the entire URL. Currently, it's `Desktop/manuscript.html#methods`. Do we need `Desktop`?

**Comment by vincerubinetti** [comment]
> I like `manuscript.html#methods` as the entire URL. Currently, it's `Desktop/manuscript.html#methods`. Do we need `Desktop`?

No, I think I added it just to make sure it was clear that it was a url. But I think everyone should recognize the chrome url bar. I'll take it out.

**Review by agitter — APPROVED** [review]
The latest updates look great to me.  I'm ready to merge once the section placement merge conflict is resolved.

We can leave the issues unchecked in #151 as a reminder to update the response to reviewers.

**Comment by dhimmel** [comment]
> The latest updates look great to me. I'm ready to merge once the section placement merge conflict is resolved.

While resolving the conflict in https://github.com/greenelab/meta-review/pull/172/commits/5d89a3d7437ad26d98daee8c53bff02e74d78613, I moved the subsection to earlier on in the Manubot section.

**Comment by agitter** [comment]
Thanks @dhimmel.  I'm ready to merge if you are.

**Comment by dhimmel** [comment]
> I'm ready to merge if you are.

Yes, go.

**Pull request merged**
